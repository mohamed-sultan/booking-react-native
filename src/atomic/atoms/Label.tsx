import React from 'react';
import {Text, StyleSheet} from 'react-native';
import  { useThemeColors } from '@store/useTheme';

const Label = ({children, style}: any) => {
  const colors = useThemeColors();

  const styles = getStyles(colors);
  return <Text style={[styles.label, style]}>{children}</Text>;
};
export default React.memo(Label);

const getStyles = (colors: any) => StyleSheet.create({
  label: {
    fontSize: 14,
    color: colors.PrimaryText,
  },
});


