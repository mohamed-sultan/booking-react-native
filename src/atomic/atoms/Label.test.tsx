import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label'; 

// Mock the theme hook
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    PrimaryText: '#000000',
  }),
}));

describe('Label', () => {
  it('renders correctly with default style', () => {
    const tree = renderer
      .create(<Label>Hello Label</Label>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom style', () => {
    const tree = renderer
      .create(
        <Label style={{ fontSize: 18, color: 'red' }}>
          Custom Styled Label
        </Label>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
