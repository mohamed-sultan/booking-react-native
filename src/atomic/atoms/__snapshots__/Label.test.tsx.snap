// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly with custom style 1`] = `null`;

exports[`Label renders correctly with default style 1`] = `null`;
